scan/tls_handshake.go

```diff
@@ -30,8 +54,9 @@ func sayHello(host string, ciphers []uint16, vers uint16) (cipherIndex int, err
 	config.MinVersion = vers
 	config.MaxVersion = vers
 	config.CipherSuites = ciphers
+	config.CurvePreferences = curves
 	conn := tls.Client(tcpConn, config)
-	serverCipher, serverVersion, err := conn.SayHello()
+	serverCipher, serverCurveType, serverCurve, serverVersion, err := conn.SayHello()
 	conn.Close()
 	if err != nil {
 		return
```

Comment:
Yep, any error from SayHello is "illegitimate", i.e.: it shouldn't be displayed or stop the scan as it's generated by the server after (predictably) invalid Client Hellos. It just needs be be served as a sentinel the show this hello was bad in some way, but the scan should be continued with other hellos.

Another possibility is adding an `ok bool` return value or something. I'm ambivalent.